libs/Wallet/src/NewWalletAccountController.cpp

```diff
@@ -0,0 +1,188 @@
+        emit accountCreatedStatus(false);
+    }
+}
+
+bool NewWalletAccountController::retrieveAndUpdateDerivedAddresses(const QString &password,
+                                                                   const WalletAccount *derivedFrom)
+{
+    assert(derivedFrom->data().derivedFrom.has_value());
+    try {
+        int currentPage = 1;
+        int foundIndex = -1;
+        int currentIndex = 0;
+        auto maxPageCount = static_cast<int>(std::ceil(static_cast<double>(m_maxDerivedAddresses)/static_cast<double>(m_derivedAddressesPageSize)));
+        std::shared_ptr<DerivedWalletAddress> foundEntry;
+        while(currentPage <= maxPageCount && foundIndex < 0) {
+            auto all = GoWallet::getDerivedAddressesForPath(StatusGo::HashedPassword(UtilsSG::hashPassword(password)),
+                                                            derivedFrom->data().derivedFrom.value(),
+                                                            Status::Constants::General::PathWalletRoot,
+                                                            m_derivedAddressesPageSize, currentPage);
+            if((currentIndex + all.size()) > m_derivedAddress.size())
+                m_derivedAddress.resize(currentIndex + all.size());
+
+            for(auto newDerived : all) {
+                auto newEntry = std::make_shared<DerivedWalletAddress>(std::move(newDerived));
+                m_derivedAddress.set(currentIndex, newEntry);
+                if(foundIndex < 0 && !newEntry->data().alreadyCreated) {
+                    foundIndex = currentIndex;
+                    foundEntry = newEntry;
+                }
+                currentIndex++;
+            }
+            currentPage++;
+        }
+        if(foundIndex > 0)
+            updateSelectedDerivedAddress(foundIndex, foundEntry);
+
+        return true;
+    } catch(const StatusGo::CallPrivateRpcError &e) {
+        return false;
+    }
+}
+
+void NewWalletAccountController::clearDerivedAddresses()
+{
+    m_derivedAddress.clear();
+}
+
+WalletAccountPtr NewWalletAccountController::findMissingAccount()
+{
+    auto accounts = GoAccounts::getAccounts();
+    // TODO: consider using a QObjectSetModel and a proxy sort model on top instead
+    auto it = std::find_if(accounts.begin(), accounts.end(), [this](const auto &a) {
+        return std::none_of(m_accounts->objects().begin(), m_accounts->objects().end(),
+                            [&a](const auto &eA) { return a.address == eA->data().address; });
+    });
+    return it != accounts.end() ? std:: make_shared<WalletAccount>(*it) : nullptr;
+}
+
+NewWalletAccountController::AccountsModel::ObjectContainer
+NewWalletAccountController::filterMainAccounts(const AccountsModel &accounts)
+{
+    AccountsModel::ObjectContainer out;
+    const auto &c = accounts.objects();
+    std::copy_if(c.begin(), c.end(), std::back_inserter(out), [](const auto &a){ return a->data().isWallet; });
```

Comment:
Hmm this filtering in general should really be part of some proxy model

Reply:
Agree, see the comment on `m_mainAccounts`: `/// \todo make it a proxy filter on top of \c m_accounts`